src/openrct2/ride/Vehicle.cpp

```diff
@@ -7280,26 +7280,41 @@ void Vehicle::UpdateAnimationAnimalFlying()
 void Vehicle::UpdateAdditionalAnimation()
 {
     uint8_t targetFrame{};
-    uint8_t curFrame{};
-    uint32_t eax{};
-
     auto vehicleEntry = Entry();
     if (vehicleEntry == nullptr)
     {
         return;
     }
+    // similar animation styles all update the same
     switch (vehicleEntry->animation)
     {
-        case VEHICLE_ENTRY_ANIMATION_MINITURE_RAILWAY_LOCOMOTIVE: // loc_6D652B
+        case VEHICLE_ENTRY_ANIMATION_MINITURE_RAILWAY_LOCOMOTIVE:
+        case VEHICLE_ENTRY_ANIMATION_SWAN:
+        case VEHICLE_ENTRY_ANIMATION_CANOES:
+        case VEHICLE_ENTRY_ANIMATION_ROW_BOATS:
+        case VEHICLE_ENTRY_ANIMATION_WATER_TRICYCLES:
+        case VEHICLE_ENTRY_ANIMATION_HELICARS:
             animationState += _vehicleVelocityF64E08;
-            targetFrame = (animationState >> 20) & 3;
+            // Step 1: shift the value right by 8 less than you want
```

Comment:
A good comment describes what the purpose of the code and not what each operation of the statement does.